src/engine/GroupBy.cpp

```diff
@@ -1045,36 +1044,39 @@ void GroupBy::computeGroupByForHashMapOptimization(
         sparqlExpression::ExpressionResult expressionResult =
             exprChildren[0]->evaluate(&evaluationContext);
 
-        auto visitor = [&currentBlockSize, &evaluationContext, &hashEntries,
-                        &aggregate]<sparqlExpression::SingleExpressionResult T>(
-                           T&& singleResult) mutable {
-          auto generator = sparqlExpression::detail::makeGenerator(
-              std::forward<T>(singleResult), currentBlockSize,
-              &evaluationContext);
+        auto visitor =
+            [&currentBlockSize, &evaluationContext, &hashEntries, &aggregate,
+             &aggregationData]<sparqlExpression::SingleExpressionResult T>(
+                T&& singleResult) mutable {
+              auto generator = sparqlExpression::detail::makeGenerator(
+                  std::forward<T>(singleResult), currentBlockSize,
+                  &evaluationContext);
 
-          using NVG = sparqlExpression::detail::NumericValueGetter;
+              using NVG = sparqlExpression::detail::NumericValueGetter;
 
-          auto hashEntryIndex = 0;
-          for (const auto& val : generator) {
-            sparqlExpression::detail::NumericValue numVal =
-                NVG()(val, &evaluationContext);
+              auto hashEntryIndex = 0;
+              for (const auto& val : generator) {
+                sparqlExpression::detail::NumericValue numVal =
+                    NVG()(val, &evaluationContext);
 
-            auto& aggregateData =
-                hashEntries[hashEntryIndex]->at(aggregate.hashMapIndex_);
+                auto vectorOffset = hashEntries[hashEntryIndex];
+                auto& aggregateData =
+                    aggregationData.at(aggregate.hashMapIndex_)
```

Comment:
I think `aggregationData.at(aggregate.hashMapIndex_)` is constant for the whole loop, so you can assign a reference to it *outside* the loop.